SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialLatencyTracer.cpp

```diff
@@ -0,0 +1,360 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Utils/SpatialLatencyTracer.h"
+
+#include "Async/Async.h"
+#include "Engine/World.h"
+#include "EngineClasses/SpatialGameInstance.h"
+#include "Interop/Connection/OutgoingMessages.h"
+#include "Utils/SchemaUtils.h"
+
+#include <sstream>
+
+DEFINE_LOG_CATEGORY(LogSpatialLatencyTracing);
+
+// Stream for piping trace lib output to UE output
+class UEStream : public std::stringbuf
+{
+	int sync()
+	{
+		UE_LOG(LogSpatialLatencyTracing, Verbose, TEXT("%s"), *FString(str().c_str()));
+		str("");
+		return std::stringbuf::sync();
+	}
+};
+
+namespace
+{
+	UEStream Stream;
+}
+
+const TraceKey USpatialLatencyTracer::ActiveTraceKey = 0;
+const TraceKey USpatialLatencyTracer::InvalidTraceKey = -1;
+
+USpatialLatencyTracer::USpatialLatencyTracer()
+{
+#if TRACE_LIB_ACTIVE
+	ResetWorkerId();
+#endif
+}
+
+void USpatialLatencyTracer::RegisterProject(UObject* WorldContextObject, const FString& ProjectId)
+{
+#if TRACE_LIB_ACTIVE
+	using namespace improbable::exporters::trace;
+
+	StackdriverExporter::Register({ TCHAR_TO_UTF8(*ProjectId) });
+
+	std::cout.rdbuf(&Stream);
```

Comment:
Possibly noob question as I'm not that familiar with the io libraries - why do we change the buffer of cout and cerr instead of defining some ostream object ourselves?

Reply:
Is it that the trace library is using cout? Are we worried this might conflict with other stuff?

Reply:
Trace library is using cout, yes

Reply:
Is it possible for us to change that to give them an ostream instead or is there some reason it needs to be via the globals that everyone uses (possibly not something to deal with right now but to ask worker about).

Reply:
Internally the worker library uses opencensus (https://opencensus.io/) and that uses cout

Reply:
Fair enough